### PR TITLE
lnav: compile with libarchive

### PR DESCRIPTION
**Pull request by tstack**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/lnav.rb [hunk 1]
```diff
@@ -40,8 +42,9 @@ def install
     system "./autogen.sh" if build.head?
     system "./configure", "--disable-dependency-tracking",
```
Comment (carlocab) [inline Formula/lnav.rb:43]:
```suggestion
    system "./configure", "--disable-dependency-tracking",
                          "--disable-debug",
                          "--disable-silent-rules",
```
Let's try to be consistent with how we build other formulae.

**Review by carlocab — COMMENTED** [review]
This also seems to be building its own `yajl` and `pugixml`. Can we use Homebrew versions of those?

File: Formula/lnav.rb (hunk 1 above)
Comment (bayandin) [reply Formula/lnav.rb:43]: Or we can use `*std_configure_args`, it includes the following: `--disable-debug`, `--disable-dependency-tracking`, `--prefix=#{prefix}`, `--libdir=#{lib}`

File: Formula/lnav.rb [hunk 2]
```diff
@@ -40,8 +42,9 @@ def install
...
                           "--prefix=#{prefix}",
```
Comment (carlocab) [inline Formula/lnav.rb:42]:
```suggestion
    system "./configure", *std_configure_args,
```
As suggested above.

File: Formula/lnav.rb [hunk 3]
```diff
@@ -38,11 +40,12 @@ class Lnav < Formula
...
-    system "./configure", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}",
-                          "--with-sqlite=#{Formula["sqlite"].opt_prefix}",
-                          "--with-readline=#{Formula["readline"].opt_prefix}"
-    system "make", "install"
+    system "./configure", *std_configure_args,
+                          "--with-sqlite3=#{Formula["sqlite"].opt_prefix}",
+                          "--with-readline=#{Formula["readline"].opt_prefix}",
+                          "--with-libarchive=#{Formula["libarchive"].opt_prefix}",
+                          "LDFLAGS=#{Formula["libarchive"].opt_prefix}/lib"
```
Comment (carlocab) [inline Formula/lnav.rb:47]:
```suggestion
                          "LDFLAGS=-L#{Formula["libarchive"].opt_prefix}/lib"
```
?

**Comment by tstack** [comment]
> This also seems to be building its own yajl and pugixml. Can we use Homebrew versions of those?

There's no support for using external versions of these libraries at the moment.

**Comment by carlocab** [comment]
I see. That's fine then. Can you squash your commits together into your first one? This should be good to merge after that.

**Review by carlocab — APPROVED** [review]
Thanks, @tstack!

**Pull request closed without merging**